server/src/main/java/org/elasticsearch/cluster/health/ClusterIndexHealth.java

```diff
@@ -171,13 +171,7 @@ public ClusterIndexHealth(final StreamInput in) throws IOException {
         initializingShards = in.readVInt();
         unassignedShards = in.readVInt();
         status = ClusterHealthStatus.readFrom(in);
-
-        int size = in.readVInt();
-        shards = Maps.newMapWithExpectedSize(size);
-        for (int i = 0; i < size; i++) {
-            ClusterShardHealth shardHealth = new ClusterShardHealth(in);
-            shards.put(shardHealth.getShardId(), shardHealth);
-        }
+        shards = in.readMapValues(ClusterShardHealth::new, ClusterShardHealth::getShardId);
```

Comment:
I think we should use `out.writeMapValues` everywhere that we use `in.readMapValues` if possible - it's good to keep the de/serialization methods in sync. As it stands we aren't using `writeMapValues` in any production code.